### DISCUSSION TITLE
Retrieve information from a collection based on it's right

### DISCUSSION THREAD

**Discussion by CogalTek** [pocketbase/pocketbase#3762] [Q&A] [answered]
Hello, I would like to retrieve in my collection a whole list of all the users who are part of a group but only if my accreditation level is greater than 2, I don't know if I should do it with the filters or directly in the code, I have a little difficulty understanding,

What do you recommend ?

**Comment by CogalTek** [comment]
I'm trying to understand with this documentation:

https://github.com/pocketbase/js-sdk/blob/master/README.md

**Reply by ganigeorgiev** [reply]
I think the discussion need some context as I currently don't understand what exactly is the question, nor what you are trying to do.

Could you elaborate a little more on your data structure and provide some info about your use case and the expected outcome?

**Reply by CogalTek** [reply]
I have a group, users and participant collection, when I want to add participants to a group I create an entry in it I put a foreign key on the group and another on the user, as well as a level (int) ranging from 0 to 3, above 2 I want the person to be able to see all the participants and modify them (addition, deletion),

The problem is that I don't understand how to search pocketbase for all the entries for a given group by checking that I have access (level>2)

<img width="613" alt="Capture d’écran 2023-11-16 à 09 52 08" src="https://github.com/pocketbase/pocketbase/assets/91875582/5d7ac854-03ab-4f95-b2a8-f0e05f08a7e9">

<img width="1122" alt="Capture d’écran 2023-11-16 à 09 51 29" src="https://github.com/pocketbase/pocketbase/assets/91875582/43fc059f-3509-4dff-9b29-15239a9e9002">

<img width="230" alt="Capture d’écran 2023-11-16 à 09 50 56" src="https://github.com/pocketbase/pocketbase/assets/91875582/4b866612-5ed3-45f6-b866-363b322721cb">

**Reply by CogalTek** [reply]
<img width="665" alt="Capture d’écran 2023-11-16 à 09 53 52" src="https://github.com/pocketbase/pocketbase/assets/91875582/23185907-0d26-4bd2-887a-810dd40f7d82">
<img width="665" alt="Capture d’écran 2023-11-16 à 09 54 04" src="https://github.com/pocketbase/pocketbase/assets/91875582/1649dd62-c190-4577-8007-45edb34e91dc">

**Reply by ganigeorgiev** [reply]
I'm sorry, I'm failing to understand the question and the desired behavior.

In general, if you want to filter by a relation field, you can use dot-notation (eg. `group.something = 123`).
Filters can be applied both server-side (aka. the API rules in the collection settings) and client-side (aka. as part of your request).

**Reply by ganigeorgiev** [reply]
Also the `level` field doesn't seem to be a number, but a string. If you want to restrict access to a specific participant action you can define an API rule like: `@request.auth.id = user && (level = "modo" || level = "owner")` (_but again, I don't understand what actually is the expected behavior so treat it just as an example_).

**Reply by CogalTek** [reply]
ok thank you

**Reply by CogalTek** [reply]
```Json
[
    {
        "collectionId": "b4c11cnxozkggtu",
        "collectionName": "participant",
        "created": "2023-11-16 09:08:27.087Z",
        "expand": {
            "user": {
                "avatar": "",
                "collectionId": "_pb_users_auth_",
                "collectionName": "users",
                "created": "2023-11-10 10:27:18.662Z",
                "email": "mercure@gestionbat.fr",
                "emailVisibility": false,
                "id": "o3qhgcplfmjy4d8",
                "name": "",
                "updated": "2023-11-10 10:27:18.662Z",
                "username": "users94494",
                "verified": true
            }
        },
        "group": "lx5sxqv1tr6is0f",
        "id": "fhpi1m6nv9tqw8v",
        "level": "owner",
        "updated": "2023-11-16 09:08:27.087Z",
        "user": "o3qhgcplfmjy4d8"
    },
    {
        "collectionId": "b4c11cnxozkggtu",
        "collectionName": "participant",
        "created": "2023-11-16 09:52:38.769Z",
        "group": "lx5sxqv1tr6is0f",
        "id": "uathglpmr7mn3kj",
        "level": "user",
        "updated": "2023-11-16 09:52:38.769Z",
        "user": "yypepviq8ohlfyi"
    },
    {
        "collectionId": "b4c11cnxozkggtu",
        "collectionName": "participant",
        "created": "2023-11-16 10:01:59.275Z",
        "group": "lx5sxqv1tr6is0f",
        "id": "jd8qf6uzw91bnue",
        "level": "user",
        "updated": "2023-11-16 10:01:59.275Z",
        "user": "4k5s9efu5upyfiw"
    }
]
```


finally I found what I wanted, the problem is that in my query I have expands user but I only have access to the data that corresponds to me, not to that of other users, is it possible that it need to do something in user to authorize reading via participant expand?


<img width="921" alt="Capture d’écran 2023-11-16 à 11 09 49" src="https://github.com/pocketbase/pocketbase/assets/91875582/c6b69ab6-0899-4696-840c-8cad3ea8805d">

**Reply by CogalTek** [reply, marked as answer]
Only the relations that the request client can View (aka. satisfies the relation collection's View API Rule) will be expanded.
Nested relations are supported via dot-notation and up to 6-levels depth.

https://pocketbase.io/docs/working-with-relations/
